src/MySqlConnector/Protocol/Payloads/OkPayload.cs

```diff
@@ -79,7 +75,28 @@ internal sealed class OkPayload
 						switch (kind)
 						{
 							case SessionTrackKind.Schema:
-								newSchema = Encoding.UTF8.GetString(reader.ReadLengthEncodedByteString());
+								context.Database = Encoding.UTF8.GetString(reader.ReadLengthEncodedByteString());
+								break;
+
+							case SessionTrackKind.SystemVariables:
+								var systemVariableOffset = reader.Offset + dataLength;
+								do
+								{
+									var variableSv = Encoding.ASCII.GetString(reader.ReadLengthEncodedByteString());
+									var lenSv = reader.ReadLengthEncodedIntegerOrNull();
+									var valueSv = lenSv == -1
+										? null
+										: Encoding.ASCII.GetString(reader.ReadByteString(lenSv));
+									switch (variableSv)
+									{
+										case "character_set_client":
+											context.ClientCharset = valueSv;
+											break;
+										case "connection_id":
+											context.ConnectionId = Convert.ToInt32(valueSv, CultureInfo.InvariantCulture);
```

Comment:
We can also use `Utf8Parser` to convert UTF-8 bytes directly to an `int` without allocating a temporary string.